src/logics/sonne/logic.repay.ts

```diff
@@ -0,0 +1,69 @@
+import { CErc20Immutable__factory } from './contracts';
+import * as common from '@protocolink/common';
+import * as core from '@protocolink/core';
+import { supportedChainIds, toCToken, underlyingTokens } from './configs';
+
+export type RepayLogicTokenList = common.Token[];
+
+export type RepayLogicParams = core.RepayParams;
+
+export type RepayLogicFields = core.RepayFields;
+
+export class RepayLogic
+  extends core.Logic
+  implements core.LogicTokenListInterface, core.LogicOracleInterface, core.LogicBuilderInterface
+{
+  static id = 'repay';
+  static protocolId = 'sonne';
+  static readonly supportedChainIds = supportedChainIds;
+
+  getTokenList() {
+    const tokens = underlyingTokens[this.chainId];
+
+    const tokenList: RepayLogicTokenList = [];
+    for (const token of tokens) {
+      if (token.isWrapped) {
+        tokenList.push(token.unwrapped);
+      }
+      tokenList.push(token);
+    }
+    return tokenList;
+  }
+
+  async quote(params: RepayLogicParams) {
+    const { borrower, tokenIn } = params;
+
+    const cToken = toCToken(this.chainId, tokenIn.wrapped);
+    const borrowBalanceWei = await CErc20Immutable__factory.connect(
+      cToken.address,
+      this.provider
+    ).callStatic.borrowBalanceCurrent(borrower);
+    const input = new common.TokenAmount(tokenIn).setWei(borrowBalanceWei);
```

Comment:
Should we increase 1 bps in `quote()` and handle `repayAll` in `build` similar to how compound-v3 does?

Reply:
yeah. I think it is necessary. But do we need to revise compound-v2 borrow as well?

Reply:
Since we have no plans to integrate compound-v2 into production and it is being deprecated, we may remove it in the future. So don't need to fix it.